agent/local.go

```diff
@@ -76,8 +91,23 @@ type localState struct {
 
 // NewLocalState creates a  is used to initialize the local state
 func NewLocalState(c *Config, lg *log.Logger) *localState {
+	lc := localStateConfig{
+		ACLToken:            c.ACLToken,
+		AEInterval:          c.AEInterval,
+		AdvertiseAddr:       c.AdvertiseAddr,
+		CheckUpdateInterval: c.CheckUpdateInterval,
+		Datacenter:          c.Datacenter,
+		NodeID:              c.NodeID,
+		NodeName:            c.NodeName,
+		TaggedAddresses:     map[string]string{},
+		TokenForAgent:       c.GetTokenForAgent(),
```

Comment:
#3225 will formalize this approach a bit more to make decoupling easier. See the dns server change.